### PR TITLE
Use block indent inside raw strings

### LINKED ISSUE #60946: Raw string completion should match leading indentation of previous line
Version: VS 17.2 preview 4

```cs
var str = """
    void M()
    {
        var x = 1;$$ // hit enter here
    }
    """;
```

Expected behavior:
```cs
var str = """
    void M()
    {
        var x = 1;
        $$ // cursor is here
    }
    """;
```

Actual behavior:
```cs
var str = """
    void M()
    {
        var x = 1;
    $$ // cursor is here
    }
    """;
```

Similar to #60945, except we don't have an "opening" token at the end of the line, and the desired behavior is to match the indentation rather than increase it.

### PR DESCRIPTION
**Pull request by sharwell**
Fixes #60946

### CODE REVIEW

File: src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/Indentation/CSharpIndentationService.Indenter.cs [hunk 1]
```diff
@@ -84,9 +84,30 @@ private static IndentationResult GetIndentationBasedOnToken(Indenter indenter, S
...
-                var nonWhitespaceOffset = endLine.GetFirstNonWhitespaceOffset();
-                Contract.ThrowIfNull(nonWhitespaceOffset);
-                return new IndentationResult(indenter.LineToBeIndented.Start, nonWhitespaceOffset.Value);
+                var minimumOffset = endLine.GetFirstNonWhitespaceOffset();
+                Contract.ThrowIfNull(minimumOffset);
+
+                // If possible, indent to match the indentation of the previous non-whitespace line contained in the
+                // same raw string. Otherwise, indent to match the ending line of the raw string.
+                var startLine = sourceText.Lines.GetLineFromPosition(token.SpanStart);
+                for (var currentLineNumber = indenter.LineToBeIndented.LineNumber - 1; currentLineNumber >= startLine.LineNumber + 1; currentLineNumber--)
```
Comment (CyrusNajmabadi) [inline src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/Indentation/CSharpIndentationService.Indenter.cs:93]: what if currentLineNumber ends up either being negative, or gets the line the raw token starts on?  should we ensure that we start at Max(LineNumber - 1, startLine + 1) ?

Comment (sharwell) [reply src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/Indentation/CSharpIndentationService.Indenter.cs:93]: It stops at `startLine.LineNumber + 1`, case is covered by `RawString1`.

Comment (CyrusNajmabadi) [reply src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/Indentation/CSharpIndentationService.Indenter.cs:93]: can you also add tests that have the raw string on line 0 (which would be legal in top level statements).

**Review by CyrusNajmabadi — APPROVED** [review]

File: src/EditorFeatures/CSharpTest/Formatting/Indentation/SmartIndenterTests.cs [hunk 2]
```diff
@@ -1389,6 +1389,123 @@ void Test()
...
+        [Trait(Traits.Feature, Traits.Features.SmartIndent)]
+        public void RawString6()
+        {
+            var code = @"var goo = """"""
+
+    """""";
+";
+            AssertSmartIndent(
+                code,
+                indentationLine: 1,
```
[the remark this answers is not in the document]
Comment (sharwell) [reply src/EditorFeatures/CSharpTest/Formatting/Indentation/SmartIndenterTests.cs:1505]: ➡️ Amended this commit to add this test case

File: src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/Indentation/CSharpIndentationService.Indenter.cs [hunk 3]
```diff
@@ -84,26 +84,81 @@ private static IndentationResult GetIndentationBasedOnToken(Indenter indenter, S
...
+                }
+
+                return indenter.GetIndentationOfLine(endLine);
             }
 
             // case 1: $"""$$
             //          """
             // case 2: $"""
             //          text$$
             //          """
```
Comment (CyrusNajmabadi) [inline src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/Indentation/CSharpIndentationService.Indenter.cs:117]: update cases.

**Review by CyrusNajmabadi — APPROVED** [review]

File: src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/Indentation/CSharpIndentationService.Indenter.cs (hunk 3 above)
Comment (sharwell) [reply src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/Indentation/CSharpIndentationService.Indenter.cs:117]: ➡️ Added third case that was previously missing

**Pull request merged**